### ISSUE TITLE
Download and process GMTED Elevation

### ISSUE THREAD

**Issue by eva0marques** [Spatiotemporal-Exposures-and-Toxicology/NRT-AP-Model#207]
- [ ] Download data with download_gmted_data.R fonction
Remark: resolution options are 30 arc-seconds (about 1 kilometer), 15 arc-seconds (about 500 meters), and 7.5 arc- seconds (about 250 meters). How can we pretend to reach 100m statistics?
- [ ] Create a script to extract elevation within buffers (and eventually other terrain covariates?)
- [ ] Create the associates test
- [ ] Provide a vignette to show the extraction on AQS monitors data

**Comment by eva0marques** [comment]
@Spatiotemporal-Exposures-and-Toxicology @mitchellmanware 

The download_gmted_data.R function is based on GMTED product, which already provides statistics at ~250m, ~500m, ~1km. In the covariates sheet, it is mandatory to have 100m, 1km and 10km statistics. 

My question: 
- do we change the covariates sheet to focus only on 250m, 500m, 1km already available resolutions?
- should I download raw 30m dataset and compute myself the 100m and 1km statistics (it should not be a good option for some of the statistics that I do not know about: systematic subsample, breakline emphasis)? 
- do I ignore 100m resolution and compute a 10km statistic from the statistics already available (confusing I know...)?

Personally, I feel like 1st option is the simplest, except if PM2.5 experts say that 100m and 10km resolution are really important.

**Comment by Spatiotemporal-Exposures-and-Toxicology** [comment]
@eva0marques I say keep it simple and do (1). In an ideal world every covariate with buffer sizes would have the same buffers, but obviously that is not a requirement.

**Comment by mitchellmanware** [comment]
- [ ] Download data with `download_gmted()` [January 25- in progres]
- [ ] `import_gmted`
  - [ ]  Write unit tests
  - [ ] Create function
- [ ] `covar_gmted`
  - [ ] Write unite tests
  - [ ] Create function
- [ ] Calculate covariates at sites

**Comment by mitchellmanware** [comment]
Elevation covariates calculated from GMTED data. Covariates include "Breakline Emphasis", "Systematic Subsample", "Median Statistic", "Minimum Statistic", "Mean Statistic", "Maximum Statistic", and "Standard Deviation Statistic" elevation values at 7.5, 15, and 30 arc second resolutions, respectively (21 covariates total). As discussed above, I did not buffer the sites to the standard 100, 1000, and 10000 meters, but rather used the variation in resolutions in the raw data. If we want to re-run with standard buffers it should not take too long.

The data set is huge, so I have zipped the raw data folders into a single zip file `/ddn/gs1/group/set/Projects/NRT-AP-Model/input/gmted/gmted_raw.zip` (zipping now; should be visible tomorrow). The raw data is ESRI ArcGRID format, so when read with `terra` the entire folder is the file path. Example:
```
> terra::rast("be15_grd/")
class       : SpatRaster 
dimensions  : 33600, 86400, 1  (nrow, ncol, nlyr)
resolution  : 0.004166667, 0.004166667  (x, y)
extent      : -180.0001, 179.9999, -56.00014, 83.99986  (xmin, xmax, ymin, ymax)
coord. ref. : lon/lat WGS 84 (EPSG:4326) 
source      : be15_grd 
name        : be15_grd 
min value   :     -872 
max value   :     8747 
> 
```
The folder `be15_grd/` contains several sub files, but to read the data set in its entirety it is advised to read the folder with `terra`. 

Calculated covariates are at `/ddn/gs1/group/set/Project/NRT-AP-Model/output/NRTAP_Covar_GMTED.rds`. Let me know if you have any questions.
